osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs

```diff
@@ -108,28 +119,48 @@ public DrawableStoryboardAnimation(StoryboardAnimation animation)
             Animation = animation;
             Origin = animation.Origin;
             Position = animation.InitialPosition;
-            Loop = animation.LoopType == AnimationLoopType.LoopForever;
 
             LifetimeStart = animation.StartTime;
             LifetimeEnd = animation.EndTime;
         }
 
         [BackgroundDependencyLoader]
-        private void load(IBindable<WorkingBeatmap> beatmap, TextureStore textureStore)
+        private void load(IBindable<WorkingBeatmap> beatmap)
         {
+            InternalChild = drawableTextureAnimation = new TextureAnimation
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Loop = Animation.LoopType == AnimationLoopType.LoopForever
+            };
+
             for (var frame = 0; frame < Animation.FrameCount; frame++)
             {
                 var framePath = Animation.Path.Replace(".", frame + ".");
+                texturePathsRaw.Add(Path.GetFileNameWithoutExtension(framePath));
 
                 var path = beatmap.Value.BeatmapSetInfo.Files.Find(f => f.Filename.Equals(framePath, StringComparison.OrdinalIgnoreCase))?.FileInfo.StoragePath;
-                if (path == null)
-                    continue;
-
-                var texture = textureStore.Get(path);
-                AddFrame(texture, Animation.FrameDelay);
+                texturePaths.Add(path);
             }
 
             Animation.ApplyTransforms(this);
         }
+
+        protected override void SkinChanged(ISkinSource skin, bool allowFallback)
+        {
+            base.SkinChanged(skin, allowFallback);
+
+            drawableTextureAnimation.ClearFrames();
+
+            for (var frame = 0; frame < Animation.FrameCount; frame++)
+            {
+                var texture = skin?.GetTexture(texturePathsRaw[frame]) ?? storyboardTextureStore?.Get(texturePaths[frame]);
```

Comment:
Doing things this way causes storyboard sprites to unnecessarily reload every time a skin changes. We probably want to bring back the old lookup logic in `load()`, but if the lookup fails, just fall back to a nested `SkinnableDrawable` which should handle the skin-changing logic cleanly on its own.

Same thing goes for sprites.

Reply:
We can't do that with `SkinnableDrawable`. `SkinnableDrawable` is for loading whole gameplay components (like hitcircles) if I'm not mistaken. We need to be able to load individual skin textures. Default font (the one on the hitcircles) as an animation for example. Also, it needs to be done during skin reload because stable replaces missing frames from skin with storyboard ones.

Reply:
You should be able to nest one. Also using `SkinnableSprite` for simplicity.

For individual textures in an animation, I'd argue we shouldn't bother supporting this.

Reply:
Somehow I missed SkinnableSprite. My bad.